x-pack/plugins/notifications/server/services/types.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export interface EmailService {
+  sendPlainTextEmail(payload: PlainTextEmail): Promise<void>;
+}
+
+export interface EmailServiceStart {
+  isEmailServiceAvailable(): boolean;
+  getEmailService(): EmailService;
+}
+
+export interface IEmailServiceProvider<T, U> {
+  setup(setupDeps: T): void;
+  start(startDeps: U): EmailServiceStart;
+}
+
+interface RelatedSavedObject {
+  id: string;
+  type: string;
+  namespace: string;
```

Comment:
I think this should be `namespaces: string[];` because a saved object can belong to multiple spaces.

Reply:
🤔 we have a discrepancy here. The actions plugin API [only contemplates a single `namespace: string`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/actions/server/lib/related_saved_objects.ts#L15) for each of the "relatedSavedObjects". Isn't there a way that you can choose the most relevant one? 😝 

A possible "solution" would be for the notifications plugin to split the related SO into multiple related SOs if they belong to multiple spaces. 
On the other hand, we could update the schema of RelatedSavedObjects, but I'm afraid this would have a bunch of side effects that we'd have to adjust to.

WDYT @ymao1

Reply:
Ah yea, when we added the `relatedSavedObjects`, the concept of saved objects in multiple namespaces did not exist  yet. It makes sense to update to `namespaces`. I can work on a PR for this.

Reply:
Ying is working on a PR to support storing a list of _spaces_ for each of the related SOs.
In the meantime, I'll split each related SO into multiple ones (one per space) behind the scenes, so that it is transparent to the callers.

Then, when `'actions'` plugin [enhancement](https://github.com/elastic/kibana/pull/144111) is available, I'll update the code to make use of the _namespace list_. This way we're not blocked and we can continue aiming at 8.6.0

Reply:
Hum, I'm not sure to get it. Yea, shareable SOs can be in multiple spaces. However they are always accessed from **one** given space.

 What are these `RelatedSavedObject` going to be used for? Generate a link to a given object? if so, listing all of the object's space doesn't seem to bring value, as the object will have to be 'used' in the context of a single space.

Reply:
Thanks for raising the point @pgayvallet !
We discussed and agreed that for this use case we will log only the `spaceId` of the space where the `Case` is assigned.
We'll switch over to logging multiple `spaceIds` when we face a use case that leaves us no alternative.